### PR TITLE
docs(how-to-guides): add training docs for centerpoint

### PR DESCRIPTION
**Pull request by kaancolak**
## Description

Prepare documentation about training lidar CenterPoint model 

### Blocked by:
- https://github.com/autowarefoundation/autoware.universe/pull/5570
- https://github.com/autowarefoundation/mmdetection3d/pull/1

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kaancolak** [comment]
Hello @yukke42 -san, @yukkysaito -san, @mitsudome-r -san 

I prepared an overview section of lidar CenterPoint training under Autoware documentation and training steps under the "lidar_centerpoint" package. 

Before we release the fork of the mmdetection3D and sample_dataset(Tier4 Format) under Autoware Foundation, please review the dataset and fork of the mmdetection3D repository. I am waiting for your feedback. 

Fork of the mmdetection3D (dev-1.x-autoware branch): 
https://github.com/open-mmlab/mmdetection3d/compare/main...kaancolak:mmdetection3d:dev-1.x-autoware

Sample Dataset(Tier4 Format)
https://drive.google.com/drive/folders/15VeeXje1La0FwUyt1hU8pKXXfcCvRyiJ

File: docs/how-to-guides/training-machine-learning-models/training-models.md [hunk 1]
```diff
@@ -28,8 +28,17 @@ an example dataset containing three distinct classes (green, yellow, red), which
...
-Training traffic light detection model and lidar CenterPoint model will be added there.
+The CenterPoint 3D object detection model within the Autoware has been trained using the **[open-mmlab/mmdetection3d](https://github.com/open-mmlab/mmdetection3d)** repository. CenterPoint
+implementation of mmdetection3d uses 10 input feature for PointPillars voxel encoder. However, Autoware employs 9 pillar features for CenterPoint, in accordance with the original research paper. Therefore, we have
+fork the original repository and made the necessary code modifications to support the use of 9 PointPillar input features. The forked repository can be found **[here](https://github.com/autowarefoundation/mmdetection3d)**.
 
--->
+To train custom CenterPoint models and convert them into ONNX format for deployment in Autoware, please refer to the instructions provided in the README file included with the
+**"lidar_centerpoint"** package. These instructions will provide a step-by-step guide for training CenterPoint model.
+
+In order to assist you with your training process, we have also included an example dataset in the Tier4 dataset format.
```
Comment (yukkysaito) [inline docs/how-to-guides/training-machine-learning-models/training-models.md:40]:
```suggestion
In order to assist you with your training process, we have also included an example dataset in the TIER IV dataset format.
```

**Comment by mitsudome-r** [comment]
@kaancolak FYI, I have created the fork of MMdetection under Autowarefoundation GitHub: https://github.com/autowarefoundation/mmdetection3d.
You should have access to create branch as well so feel free to port your work to the fork.

**Comment by miursh** [comment]
@kaancolak 
I sincerely apologize for the delay in reviewing. 
I have noted some comments and change requests regarding the dataset:

1. A smaller file size for the T4 dataset is preferable, as the current size is leading to prolonged download times.
2. The directory structure for each dataset should adhere to the guidelines outlined in [this document](https://github.com/tier4/tier4_perception_dataset/blob/main/docs/t4_format_3d_detailed.md#directory-structure). Specifically, there should be an 'annotation/data/input_bag' directory directly under 'T4_dataset_ID' (or 'scene_name').
3. If the annotation frequency (Hz) is lower than that of the data, frames that were not annotated should not be included in sample.json. Additionally, for those frames, the is_key_frame in sample_data.json should be set to false. This might be due to our tools bug.
5. It appears that the calibration, especially between CAM_BACK and base_link, is not well-aligned. Is this OK?

**Comment by kaancolak** [comment]
Hi @miursh -san, thank you for your feedback. Following that, I plan to filter out non-annotated frames from the dataset. This will reduce the total size from 32 GB to approximately 6 GB and eliminate the need to deal with keyframes.

I will also inspect the calibration of the back camera. If there seems any issue, I can remove them from the dataset.

**Comment by kaancolak** [comment]
Miura-san (@miursh), 

I reloaded the dataset based on your feedback. Can you take a look and give feedback?

The total size decreased by around half of the previous version.  (~13 GB)

The back camera is mounted on the rear glass of the vehicle and that part can be opened and closed as a car luggage. Somehow the calibration seems distorted, especially at a far distance. I removed it from the dataset.

Dataset link: https://drive.google.com/drive/folders/1qpts1HlOT3PdvrV_dRd1Q9syY2fhl8pO

**Comment by miursh** [comment]
@kaancolak I apologize for the delay in responding as I was taking some day off. 
The updated dataset looks good to me! Thank you for the change!

**Comment by kaancolak** [comment]
Thank you so much @miursh -san. I will ask to Fatih upload the dataset to AWS.

File: docs/how-to-guides/training-machine-learning-models/training-models.md [hunk 2]
```diff
@@ -28,8 +28,17 @@ an example dataset containing three distinct classes (green, yellow, red), which
 
 Detailed instructions for training the traffic light classifier model can be found **[here](https://github.com/autowarefoundation/autoware.universe/blob/main/perception/traffic_light_classifier/README.md)**.
 
-<!--
+## Training CenterPoint 3D object detection model
 
-Training traffic light detection model and lidar CenterPoint model will be added there.
+The CenterPoint 3D object detection model within the Autoware has been trained using the **[open-mmlab/mmdetection3d](https://github.com/open-mmlab/mmdetection3d)** repository. CenterPoint
+implementation of mmdetection3d uses 10 input features for the PointPillars voxel encoder. However, Autoware employs 9 pillar features for CenterPoint, in accordance with the original research paper. Therefore, we have
```
Comment (Shin-kyoto) [inline docs/how-to-guides/training-machine-learning-models/training-models.md:34]: > 10 input features

This expression is ambiguous. What is 10 input features?

Comment (Shin-kyoto) [inline docs/how-to-guides/training-machine-learning-models/training-models.md:34]: These specific change log should be written in https://github.com/open-mmlab/mmdetection3d

**Review by Shin-kyoto — CHANGES_REQUESTED** [review]

File: docs/how-to-guides/training-machine-learning-models/training-models.md (hunk 2 above)
Comment (kaancolak) [reply docs/how-to-guides/training-machine-learning-models/training-models.md]: @Shin-kyoto -san, Thank you, I changed the paragraph a little bit to make the documentation more clear.

Comment (Shin-kyoto) [reply docs/how-to-guides/training-machine-learning-models/training-models.md]: @kaancolak 
Thanks! Can you move the reason why we have to make the training repository for centepoint to https://github.com/autowarefoundation/mmdetection3d ?

Comment (kaancolak) [reply docs/how-to-guides/training-machine-learning-models/training-models.md]: @Shin-kyoto  -san,  I refactored the paragraph, moved the first part to the ReadMe file of the AutowareCenterPoint project, and linked with it. 

https://github.com/autowarefoundation/mmdetection3d/pull/1/files#diff-c1d68ca9b975ba79c77f68690b85862e68eecfe664a6a711d02121bf4093c0b9R1

File: docs/how-to-guides/training-machine-learning-models/training-models.md [hunk 3]
```diff
@@ -28,8 +28,13 @@ an example dataset containing three distinct classes (green, yellow, red), which
...
-Training traffic light detection model and lidar CenterPoint model will be added there.
+The CenterPoint 3D object detection model within the Autoware has been trained using the **[autowarefoundation/mmdetection3d](https://github.com/autowarefoundation/mmdetection3d/blob/main/projects/AutowareCenterPoint/README.md)** repository.
 
--->
+To train custom CenterPoint models and convert them into ONNX format for deployment in Autoware, please refer to the instructions provided in the README file included with Autoware's
+**[lidar_centerpoint](https://autowarefoundation.github.io/autoware.universe/main/perception/lidar_centerpoint/)** package. These instructions will provide a step-by-step guide for training the CenterPoint model.
+
+In order to assist you with your training process, we have also included an example dataset in the TIER IV dataset format.
+This dataset contains 600 lidar frames and covers 5 classes, including 6905 cars, 3951 pedestrians, 75 cyclists, 162 buses, and 326 trucks.
+You can utilize this example dataset to facilitate your training efforts.
```
Comment (xmfcx) [inline docs/how-to-guides/training-machine-learning-models/training-models.md:40]:
```suggestion
In order to assist you with your training process, we have also included an example dataset in the TIER IV dataset format.

This dataset contains 600 lidar frames and covers 5 classes, including 6905 cars, 3951 pedestrians, 75 cyclists, 162 buses, and 326 trucks.

You can utilize this example dataset to facilitate your training efforts.
```

For readability.

**Comment by kaancolak** [comment]
Hi @beginningfan, could you also review this document?

**Review by beginningfan — APPROVED** [review]

**Pull request merged**